buffer/src/main/java/io/netty/buffer/api/DefaultCompositeBuffer.java

```diff
@@ -430,6 +427,25 @@ public void copyInto(int srcPos, Buffer dest, int destPos, int length) {
         }
     }
 
+    @Override
+    public int bytesBefore(byte needle) {
+        if (!isAccessible()) {
+            throw bufferIsClosed(this);
+        }
+        final int length = readableBytes();
+        final int start = searchOffsets(readerOffset());
```

Comment:
you can directly assign this to `I` in the for loop at it is only used within the loop